lib/designator/selection.ex

```diff
@@ -115,14 +123,18 @@ defmodule Designator.Selection do
     Stream.reject(stream, fn x -> MapSet.member?(seen_subject_ids, x) end)
   end
 
-  defp selection_subject_set_ids(all_subject_set_ids, subject_set_id) do
-    if Enum.member?(all_subject_set_ids, subject_set_id) do
+  defp selection_subject_set_ids(true, subject_set_ids, subject_set_id) do
```

Comment:
I might be a bit confused with this. We have 2 `selection_subject_set_ids` defined (similar number of params), which one is `get_streams` on line 50 expected to call?

Reply:
Oooh! I see now. Had to revisit, so if `workflow.grouped` is `true` we hit line 126 and check if `subject_set_id` is within `subject_set_ids` put in (if not return empty list if so it returns `[ subject_set_id ]`. 

If `workflow.grouped` is anything that isn't true we hit `selection_subject_set_ids` defined in line 134 which returns the `subject_set_ids`. 

Which explains what you meant in comment 
> This PR will now ensure ensure we respect `subject_set_id` param for grouped workflows and do not return data in other linked sets. For non grouped workflows this PR will ignore the `subject_set_id` param and select from all sets which is the current behaviour if the set is not found.
> 
> Thus a workflow must have grouped turned on to access per subject_set selection behaviours.